From: Jeff Mahoney <jeffm@suse.com>
Subject: workaround for vfs_link crash reported on research list
Suse-bugzilla: 60041

This is just a workaround and should be removed if it's determined that
NFS's i_op->link operation is returning success without instantiating the
dentry.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>

--- linux-2.6.11/include/linux/security.h.old	2005-03-03 16:24:56.000000000 -0500
+++ linux-2.6.11/include/linux/security.h	2005-03-03 16:25:06.000000000 -0500
@@ -1256,7 +1256,7 @@ extern int mod_unreg_security	(const cha
 #define COND_SECURITY(seop, def) 	\
 	(unlikely(security_enabled))? security_ops->seop: def
 #define COND_SECURITY_NOT_PRIVATE(inode, seop, def) 	\
-	(unlikely(security_enabled) && likely (!IS_PRIVATE(inode)) )? security_ops->seop: def
+	(unlikely(security_enabled) && likely (inode) && likely (!IS_PRIVATE(inode)) )? security_ops->seop: def
 
 #else /* CONFIG_SECURITY */
 static inline int security_init(void)
